guides/keras_cv/classification_with_keras_cv.py

```diff
@@ -0,0 +1,701 @@
+                * tf.cast(global_step - warmup_steps - hold, tf.float32)
+                / float(total_steps - warmup_steps - hold)
+            )
+        )
+    )
+
+    warmup_lr = tf.cast(target_lr * (global_step / warmup_steps), tf.float32)
+    target_lr = tf.cast(target_lr, tf.float32)
+
+    if hold > 0:
+        learning_rate = tf.where(
+            global_step > warmup_steps + hold, learning_rate, target_lr
+        )
+
+    learning_rate = tf.where(global_step < warmup_steps, warmup_lr, learning_rate)
+    return learning_rate
+
+
+class WarmUpCosineDecay(schedules.LearningRateSchedule):
+    def __init__(self, warmup_steps, total_steps, hold, start_lr=0.0, target_lr=1e-2):
+        super().__init__()
+        self.start_lr = start_lr
+        self.target_lr = target_lr
+        self.warmup_steps = warmup_steps
+        self.total_steps = total_steps
+        self.hold = hold
+
+    def __call__(self, step):
+        lr = lr_warmup_cosine_decay(
+            global_step=step,
+            total_steps=self.total_steps,
+            warmup_steps=self.warmup_steps,
+            start_lr=self.start_lr,
+            target_lr=self.target_lr,
+            hold=self.hold,
+        )
+
+        return tf.where(step > self.total_steps, 0.0, lr, name="learning_rate")
+
+
+"""
+![WarmUpCosineDecay schedule](https://i.imgur.com/YCr5pII.png)
+
+The schedule looks a as we expect.
+
+Next let's construct this optimizer:
+"""
+
+total_images = 9000
+total_steps = (total_images // BATCH_SIZE) * EPOCHS
+warmup_steps = int(0.1 * total_steps)
+hold_steps = int(0.45 * total_steps)
+schedule = WarmUpCosineDecay(
+    start_lr=0.05,
+    target_lr=1e-2,
+    warmup_steps=warmup_steps,
+    total_steps=total_steps,
+    hold=hold_steps,
+)
+optimizer = optimizers.SGD(
+    decay=5e-4,
+    learning_rate=schedule,
+    momentum=0.9,
+)
+"""
+At long last, we can now build our model and call `fit()`!
+"""
+
+backbone = keras_cv.models.EfficientNetV2Backbone.from_preset(
+    "efficientnetv2_b0",
+)
+model = keras.Sequential(
+    [
+        backbone,
+        keras.layers.GlobalMaxPooling2D(),
+        keras.layers.Dropout(rate=0.5),
+        keras.layers.Dense(101, activation="softmax"),
+    ]
+)
+
+"""
+Since the labels produced by MixUp() and CutMix() are somewhat artificial, we
+employ label smoothing to prevent the model from overfitting to artifacts of
+this augmentation process.
+"""
+loss = losses.CategoricalCrossentropy(label_smoothing=0.1)
+"""
+Let's compile our model:
+"""
+model.compile(
+    loss=loss,
+    optimizer=optimizer,
+    metrics=[
+        metrics.CategoricalAccuracy(),
+        metrics.TopKCategoricalAccuracy(k=5),
+    ],
+)
+"""
+and finally call fit().
+"""
+model.fit(
+    train_ds,
+    epochs=EPOCHS,
+    validation_data=eval_ds,
+)
+
+"""
+Congratulations!  You now know how to train a powerful image classifier from
+scratch in KerasCV.
+In practice, you'll likely want to combine transfer learning with an
```

Comment:
To add more precision here:

Depending on the availability of labeled data for your application, training from scratch may or may not be more powerful than using transfer learning in addition to the data augmentations discussed above. For smaller datasets, pretrained models generally produce high accuracy and faster convergence.